code/Modules/Input/emsc/emscInputMgr.cc

```diff
@@ -82,6 +85,124 @@ emscInputMgr::discardCallbacks() {
     emscripten_set_touchcancel_callback("#canvas", 0, true, 0);
     emscripten_set_devicemotion_callback(0, true, 0);
     emscripten_set_deviceorientation_callback(0, true, 0);
+    emscripten_set_gamepaddisconnected_callback(this, true, emscGamepadDisconnected);
+}
+
+//------------------------------------------------------------------------------
+void
+emscInputMgr::updateGamepadsAttached(int& gamepadsAttached) {
+    gamepadsAttached = emscripten_get_num_gamepads();
+
+    // TODO: Check if gamepadsAttached <= MaxNumGamepads
+    int i = 0;
+    for (; i < 4; ++i) {
```

Comment:
Please replace the '4' with the MaxNumGamepads constant that's defined in inputMgrBase.h